pineappl/src/grid.rs

```diff
@@ -1355,10 +1355,13 @@ impl Grid {
             }
         }
 
-        muf2_grid.sort_by(|a, b| a.partial_cmp(b).unwrap_or_else(|| unreachable!()));
-        muf2_grid.dedup_by(|a, b| approx_eq!(f64, *a, *b, ulps = 64));
+        // make grids unique
         x_grid.sort_by(|a, b| a.partial_cmp(b).unwrap_or_else(|| unreachable!()));
         x_grid.dedup_by(|a, b| approx_eq!(f64, *a, *b, ulps = 64));
+        mur2_grid.sort_by(|a, b| a.partial_cmp(b).unwrap_or_else(|| unreachable!()));
+        mur2_grid.dedup_by(|a, b| approx_eq!(f64, *a, *b, ulps = 64));
+        muf2_grid.sort_by(|a, b| a.partial_cmp(b).unwrap_or_else(|| unreachable!()));
```

Comment:
Actually, is not clear to me in which way `unreachable!()` is better than a plain `.unwrap()`?

Is it automatically capturing some info about the point where it is placed? Because the meaning of `.unwrap_or_else()` should be to provide further information, other than a bare panic.

Reply:
I personally use `unwrap_or_else(|| unreachable!())` if I strongly believe that `unwrap` [will never fail](https://users.rust-lang.org/t/convention-for-unwrapping-that-cant-fail/13797). The behaviour is the same as `unwrap()`, which I use instead to say: I don't care about the error handling for the time being, give me what I want. The latter simply indicates that I haven't thought about if it might possibly fail at all. If I remember correctly this also makes a small difference when linting with clippy (can't remember what difference, though).

Reply:
Ok, I see. 

But since it is not acting in a different way, this is more a personal convention. If we should use it project-wide it should be written in some guidelines, e.g. `CONTRIBUTING.md`.

However, I'd prefer not to adopt project-wide custom conventions, and stick to general Rust community conventions (i.e. everyone can do whatever he/she prefers when there is no warning by the compiler or Clippy).

For the specific topic, I'd recommend to fill even `unreachable!(msg)`  with a telling `msg`; if not anything else, it will act as a simpler tag for the error.

Reply:
If in doubt, always prefer `unwrap`. We should probably not use either of them, but for the time being it's OK to use `unwrap`.